### PR TITLE
[fix](array-type) fix the invalid format load for stream load

### PR DESCRIPTION
**Pull request by carlvinhust2012**
# Proposed changes
1. this pr is used to fix the invalid format load for stream load.
2. before the change , we will get the error when we load the invalid array format.
the origin file to load :
1       [1, 2, 3]
2       [4, 5, 6]
3       \N
4       [7, \N, 8]
5       10, 11, 12

[hugo@xafj-palo]$ sh curl_cmd.sh 
{
    "TxnId": 11035,
    "Label": "11c9f111-188e-4616-9a50-aec8b7814513",
    "TwoPhaseCommit": "false",
    "Status": "Fail",
    "Message": "Array does not start with '[' character, found '1'",
    "NumberTotalRows": 0,
    "NumberLoadedRows": 0,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 55,
    "LoadTimeMs": 7,
    "BeginTxnTimeMs": 0,
    "StreamLoadPutTimeMs": 2,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 3,
    "CommitAndPublishTimeMs": 0
}
3. after this change, we will get success and the error url which report the error line.
[hugo@xafj-palo]$ sh curl_cmd.sh 
{
    "TxnId": 11046,
    "Label": "249808ee-55f4-4c08-b671-b3d82689d614",
    "TwoPhaseCommit": "false",
    "Status": "Success",
    "Message": "OK",
    "NumberTotalRows": 5,
    "NumberLoadedRows": 4,
    "NumberFilteredRows": 1,
    "NumberUnselectedRows": 0,
    "LoadBytes": 55,
    "LoadTimeMs": 39,
    "BeginTxnTimeMs": 0,
    "StreamLoadPutTimeMs": 2,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 19,
    "CommitAndPublishTimeMs": 16,
    "ErrorURL": "http://10.81.85.89:8502/api/_load_error_log?file=__shard_3/error_log_insert_stmt_8d4130f0c18aeb0a-ad7ffd4233c41893_8d4130f0c18aeb0a_ad7ffd4233c41893"
}

the sql select result:
MySQL [example_db]> select * from array_test06;
+------+--------------+
| k1   | k2           |
+------+--------------+
|    1 | [1, 2, 3]    |
|    2 | [4, 5, 6]    |
|    3 | NULL         |
|    4 | [7, NULL, 8] |
+------+--------------+
4 rows in set (0.019 sec)

the url page show us:
"Reason: Invalid format for array column(k2). src line [10, 11, 12]; "

Issue Number: #7570

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
5. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
6. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
7. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
8. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/base_scanner.cpp [hunk 1]
```diff
@@ -481,4 +481,35 @@ void BaseScanner::_fill_columns_from_path() {
     }
 }
 
+bool BaseScanner::is_null(const Slice& slice) {
+    return slice.size == 2 && slice.data[0] == '\\' && slice.data[1] == 'N';
+}
+
+bool BaseScanner::is_array(const Slice& slice) {
+    return slice.size > 0 && slice.data[0] == '[' && slice.data[slice.size - 1] == ']';
```
Comment (cambyzju) [inline be/src/exec/base_scanner.cpp:489]:
```suggestion
    return slice.size > 1 && slice.data[0] == '[' && slice.data[slice.size - 1] == ']';
```

Comment (carlvinhust2012) [reply be/src/exec/base_scanner.cpp:489]: the size at least is 2?

Comment (carlvinhust2012) [reply be/src/exec/base_scanner.cpp]: done

File: be/src/vec/exec/vbroker_scanner.cpp [hunk 2]
```diff
@@ -95,6 +91,10 @@ Status VBrokerScanner::_fill_dest_columns(const Slice& line,
         return Status::OK();
     }
 
+    if (!check_array_format(_split_values)) {
```
Comment (xy720) [inline be/src/vec/exec/vbroker_scanner.cpp:94]: I think it is not a good implementation. We can filter the invalid row in tablet_sink.cpp.

Comment (carlvinhust2012) [reply be/src/vec/exec/vbroker_scanner.cpp:94]: As we disscuss, the 'tablet_sink.cpp' layer just can process the block which has parse the entire array. if the 'vbroker_scanner' layer can parse the array failed and return error, the 'tablet_sink.cpp' layer will not get the block to filter.

Comment (xy720) [reply be/src/vec/exec/vbroker_scanner.cpp:94]: You are right.

**Review by cambyzju — DISMISSED** [review]
LGTM

**Review by xy720 — APPROVED** [review]
LGTM

**Pull request merged**